bears/general/InvalidLinkBear.py

```diff
@@ -43,8 +43,9 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'\"`<>|\\]+|'
```

Comment:
Wow, this is complicated.
Can we please split this into sections and explain each section ?
```python
            r'(https?://'  # http or https
            r'[^.:%\s_/?#[\]@\\]+'  # Initial part of the domain
            r'\.'  # A required dot (.)
            r'(?:[^\s()%\'\"`<>|\\]+|\([^\s()%\'\"`<>|\\]*\))' # whatsit ?
            r'%?[0-9]*(?:[^\s()%\'\"`<>|\\]+|' # eh ?
            r'\([^\s()%\'\"`<>|\\]*\))*)(?<!\.)(?<!,)') # ooh
```